### PR TITLE
vep: init at 110

### PR DESCRIPTION
**Pull request by apraga**
This software has required significant work to package and need other PR to be merge beforehand : #185691 #186459 #186464 and the infamous  #186462.
The package has been formatted with nixpkgs-fmt. I was not able to run tests but this software has been used for hundred of runs in research setting.

## Description of changes

New packages for Variant Effect Predictor.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:30]: Avoid `rec`; I don't think you need it anywhere in this file.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:34]: Use `fetchFromGitHub` for this.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:42]: Use `fetchFromGitHub` for this too.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:28]: If you name these parameters `name`, `rev`, etc. you can use `inherit` in the places they are used.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:63]:
```suggestion
    doCheck = false;
```

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:85]: Why are all of these added as attributes on the derivation?

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:106]:
```suggestion
  dontBuild = true;
```

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:49]: Did you try `buildPerlPackage`, which I think would mean you don't have to override `buildPhase` and `installPhase`?

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:52]: `fetchFromGitHub`?

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:111]: What is this for?

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:32]: If using an unreleased commit, the version should be `0-unstable-YYYY-MM-DD` (see https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming).

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:122]:
```suggestion
    cp -r modules/Bio $out/${perl.libPrefix}/${perl.version}/
```

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:128]:
```suggestion
      license = lib.licenses.asl20;
```

**Review by rhendric — CHANGES_REQUESTED** [review]

**Comment by apraga** [comment]
Thanks for your review. Your changes have been incorporated. I've also switched to using the release version (with fetchzip as it's not available with fetchfromgithub) instead of git commits. Code have been formatted with nixfmt-rfc-style. The changes are in a separated commit for readability.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:94]: Add `htslib` to the top-level parameters instead of getting it from `pkgs`.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:7]: Shouldn't need this after adding `htslib`.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:16]:
```suggestion
  perl,
  makeWrapper,
```

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:74]: This should match the version string used by upstream, which looks like it's 110.0.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:98]:
```suggestion
    rev = "release/${version}";
```

**Review by rhendric — CHANGES_REQUESTED** [review]

**Comment by apraga** [comment]
Oops, somehow I forgot to clean references to `pkgs`. Thanks !

**Comment by apraga** [comment]
@rhendric If you have time to take another look at this PR, that would be great. Thanks !

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:32]:
```suggestion
      src = fetchFromGitHub {
        inherit sha256 version;
        owner = "Ensembl";
        repo = name;
        rev = "release/${version}";
      };
```

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:120]: You should add a [`platforms`](https://nixos.org/manual/nixpkgs/stable/#var-meta-platforms) attribute.

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:20]: Custom phases need to run hooks:

```suggestion
    runHook preInstall
    
    mkdir -p $out/${perl.libPrefix}/${perl.version}/
    tests=$(find modules/ -mindepth 1 -maxdepth 1 -type d | grep -v t)
    cp -r $tests $out/${perl.libPrefix}/${perl.version}/
    
    runHook postInstall
```

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [inline pkgs/by-name/ve/vep/package.nix:63]: Looks like this hash is wrong?

**Review by rhendric — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ve/vep/package.nix
Comment (apraga) [reply pkgs/by-name/ve/vep/package.nix:63]: That's weird. I've updated it but not sure why it changed (maybe github ?).

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [reply pkgs/by-name/ve/vep/package.nix:19]: :point_up:

**Comment by apraga** [comment]
@rhendric Okay, this should be better, thanks !

**Comment by apraga** [comment]
@rhendric If you have some time to look at it, it would be great :)

File: pkgs/by-name/ve/vep/package.nix
Comment (rhendric) [reply pkgs/by-name/ve/vep/package.nix:19]: I still don't see hooks here. You've added them to a copy of this script, but kept this script hookless? What's going on?

Comment (apraga) [reply pkgs/by-name/ve/vep/package.nix:19]: Thanks for catching that up. I've added the hooks.

**Comment by apraga** [comment]
@rhendric When testing the package, the binaries are not installed. Not sure how it worked before (we used it in production) but this has been fixed. If you have any comment about the wrapping, please tell me.

**Comment by apraga** [comment]
@rhendric I've merged the commits and rebase against master. Do you think we can (finally) put this PR to rest ? :)

**Review by rhendric — COMMENTED** [review]
Aside from the `pname`, nothing else leaps out at me as needing correction, but you should probably be pestering a Perl maintainer, not me. @stigtsp @zakame @marcusramberg if one of you wants to help this guy out?

File: pkgs/by-name/ve/vep/package.nix
Comment (apraga) [reply pkgs/by-name/ve/vep/package.nix]: Good catch, I've set it to `vep`, thanks for your help.

**Review by stigtsp — COMMENTED** [review]
@rhendric Looks good to me, pending ofborg eval.

I don't see any reason to hold up this PR.

**Review by rhendric — APPROVED** [review]

**Review by stigtsp — APPROVED** [review]

**Pull request merged**
